zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -888,18 +886,16 @@ public void removeRepository(String id) throws IOException {
   }
 
   public void removeNoteInterpreterSettingBinding(String user, String noteId) throws IOException {
-    synchronized (interpreterSettings) {
```

Comment:
Ah that's good point. You're right. if interpreterBindings is concurrentMap, it looks like `getNoteInterpreterSettingBinding` and `removeNoteInterpreterSettingBinding` doesn't need to have synchronized block.